portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -37,6 +41,19 @@
 
   @Override
     public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        Query query = new Query("Task").addSort("timestamp", SortDirection.DESCENDING);
+        int pageLoaded = 0;
+        PreparedQuery results = datastore.prepare(query);
+        for (Entity entity : results.asIterable()) {
```

Comment:
Since all we're doing is constructing a list from another list, it can be written quite concisely with Java streams

It's a new syntax introduced in Java 8, somewhat similar to JavaScript arrow functions. It may not work as I'm not sure on the type of `results.asIterable()` and whether that implements `Collection`, so don't worry too much about getting it working.

```java
import java.util.stream.Collectors; 

.....

List<String> comments = results.asIterable().stream()
  .map(entity -> (String) entity.getProperty("text"))
  .collect(Collectors.toList());
```